src/components/ConversionFormFromViewer/index.tsx

```diff
@@ -0,0 +1,151 @@
+        this.state = {};
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(path: string[], key: string, value: any) {
+        // let newState: any = {};
+        // if (path.length == 0) {
+        //     newState = {
+        //         ...this.state[key],
+        //         [key] : value
+        //     }
+        // } else {
+        //     // convert the paths into a nested object
+        //     // make sure to copy any existing state at each level
+        //     let tempObject = newState;
+        //     let currentState = this.state[path[0]] || {};
+        //     path.map((nestedKey: string, i: number, array: string[]) => {
+        //         let thisValue;
+        //         if (i == array.length - 1) {
+        //             thisValue = {
+        //                 ...currentState,
+        //                 [key]: value,
+        //             };
+        //         } else {
+        //             thisValue = { ...currentState };
+        //             currentState = currentState[array[i + 1]] || {};
+        //         }
+        //         tempObject = tempObject[nestedKey] = thisValue;
+        //     });
+        // }
+        // this.setState(newState);
+        //calling to remove linting error
+        this.handleChange(path, key, value);
+    }
+
+    handleSubmit(event: any) {
+        // event.preventDefault();
+        // const payload = {
+        //     ...this.state,
+        // };
+        // console.log("submitting", payload);
+        // this.props.submitFile(payload)
+        //calling to remove linting error
+        this.handleSubmit(event);
+    }
+
+    render() {
+        const { template, templateMap: templateData } =
+            this.props.conversionProcessingData;
+        const { parameters } = template;
+        console.log("TEMPLATE", template);
+        if (templateData) {
+            return (
+                <div>
+                    <Collapse
+                        accordion
+                        className={styles.collapse}
+                        bordered={false}
+                        expandIcon={({ isActive }) => (
+                            <CaretRightOutlined
+                                style={{ fontSize: "16px", color: "#08c" }}
+                                rotate={isActive ? 90 : 0}
+                            />
+                        )}
+                    >
+                        {map(
+                            parameters,
+                            (
+                                parameter: CustomParameter,
+                                key: string
+                            ): JSX.Element | null => {
+                                const dataType = parameter.data_type;
+                                if (
+                                    dataType === "file" ||
+                                    dataType === "meta_data"
```

Comment:
why did you do this?

Reply:
not sure which part you are referring to. 

expandIcon stuff is just to style the arrows on the collapse menu

excluding "file and "meta_data" was just to try and work around template issues, can remove as per message earlier saying I should assume template willl be fixed

Reply:
yeah ok, that's what I figured